### PR TITLE
hometask7

### PR DESCRIPTION
**Pull request by bogdanbasii**
Not everything works perfectly in task 1 but I hope I did the most important part.

### CODE REVIEW

File: hometask7 [hunk 1]
```diff
@@ -0,0 +1,109 @@
+# Task 1
```
Comment (klymenok) [inline hometask7]: Your code should be it `.py` file. Rename this file to `hometask7.py`

File: hometask7 [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+import json
+
+with open('json_phonebook.json') as file:
```
Comment (klymenok) [inline hometask7]: Your file will be opened during the whole program execution. This is not how it should work. Open your file only in places where you need it.

File: hometask7 [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+with open('json_phonebook.json', 'w') as file:
+    json.dump(phonebook, file)
+
+
+Task 2
+
+import time
+import os
+
+f = open('log1.txt', "w+")
```
Comment (klymenok) [inline hometask7]: It is better to use `open` as a context manager. And open the file in the place where you work with it - inside the decorator.

File: hometask7 [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+
+Task 3
+import time
+import os
+import logging
+
+try:
+    printf("This is a mistake")
+except Exception as Argument:
```
Comment (klymenok) [inline hometask7]: This is not your custom exception. It is using of Exception class

**Review by klymenok — CHANGES_REQUESTED** [review]

File: hometask7.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    if command == 'stats':
+        print(len(phonebook))
+
+
+    elif command == 'add':
+        name = input('Enter a name:')
+        if not name in phonebook.keys():
+            number = input('Enter a number:')
+            phonebook[name] = number
+            with open('json_phonebook.json', 'a') as file:
```
Comment (klymenok) [inline hometask7.py]: If you appen (mod `a`) data to the file you json structure will become invalid. Use `w+` modifier here instead

File: hometask7.py [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+                content = json.dumps(phonebook)
+                file.write(content)
+            print('New contact was added')
+        else:
+            name in phonebook.keys()
```
Comment (klymenok) [inline hometask7.py]: This line has no sense, remove it please

File: hometask7.py [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+            print("Fault. There is such name in a phonebook.")
+            with open('json_phonebook.json', 'a') as file:
```
Comment (klymenok) [inline hometask7.py]: You don't need to rewrite the file if there are no changes. Remove it

File: hometask7.py [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+
+
+# Task 3
+
+import os
+
+import time
+class MyException(Exception):
+    def __init__(self, message):
+     with open('log2.txt', 'w') as f:
```
Comment (klymenok) [inline hometask7.py]: Pay attention to indentation: your code here does not look as it should

**Review by klymenok — CHANGES_REQUESTED** [review]

File: hometask7.py (hunk 5 above)
Comment (bogdanbasii) [reply hometask7.py]: Done

File: hometask7.py (hunk 7 above)
Comment (bogdanbasii) [reply hometask7.py]: Done

File: hometask7.py (hunk 8 above)
Comment (bogdanbasii) [reply hometask7.py]: Done. Thank you

File: hometask7.py (hunk 6 above)
Comment (bogdanbasii) [reply hometask7.py]: Done

File: hometask7.py [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+
+
+# Task 3
+
+import os
+
+import time
+class MyException(Exception):
+    def __init__(self, message):
+    with open('log2.txt', 'w') as f:
```
Comment (klymenok) [inline hometask7.py]: The code is bad-formatted and therefore invalid: move the context manager block inside of the __init__ method to make it works.

File: hometask7.py [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+        break
+
+
+
+# Task 2
+
+import time
+import os
+
+f = open('log1.txt', "w+")
```
Comment (klymenok) [inline hometask7.py]: You should open this file in the place where you use it. Move this line between 78 and 79 lines. And also close the file there, between 80 and 81 lines

**Review by klymenok — CHANGES_REQUESTED** [review]

File: hometask7.py (hunk 10 above)
Comment (bogdanbasii) [reply hometask7.py]: Done

Comment (bogdanbasii) [reply hometask7.py]: Done

File: hometask7.py (hunk 9 above)
Comment (bogdanbasii) [reply hometask7.py]: Not sure that I understood you correctly: should I just indent everything after __init__ method?

Comment (klymenok) [reply hometask7.py]: No, you should mo it `inside` of the init method:
```
class MyException(Exception): # <- 0 spaces at the start of the line
    def __init__(self, message): # <- 4 spaces at the start of the line
        with open(*) as f: # <- 8 spaces at the start of the line
            # do something # <- 12 spaces at the start of the line

**Review by klymenok — APPROVED** [review]
Looks good!

**Pull request merged**